src/model/defineRelationalProperties.ts

```diff
@@ -107,17 +112,44 @@ export function defineRelationalProperties(
             },
             [],
           )
+
           log(`resolved "${relation.kind}" "${property}" to`, refValue)
 
           return relation.kind === RelationKind.OneOf
             ? first(refValue)!
             : refValue
         },
-      }
-
+      })
       return properties
     },
     {},
   )
-  Object.defineProperties(entity, properties)
+  defineNestedProperties(entity, properties, '')
+}
+
+function defineNestedProperties(
+  entity: InternalEntity<any, any>,
+  properties: any,
+  path: string,
+) {
+  for (let key in properties) {
+    const value = properties[key]
+    const nestedPath = path ? `${path}.${key}` : key
+
+    if (isRelationalProperty(value)) {
+      const nestedPathArray = nestedPath.split('.')
+      nestedPathArray.reduce((acc, curr, i) => {
+        if (i === nestedPathArray.length - 1) {
+          Object.defineProperty(acc, curr, value)
+        }
+        return acc[curr]
+      }, entity)
+    } else if (isObject(value)) {
+      defineNestedProperties(entity, value, nestedPath)
+    }
+  }
+}
+
+function isRelationalProperty(val: any) {
```

Comment:
I recommend reusing the existing `isOneOfRelation` and `isManyOfRelation` functions from here:

https://github.com/mswjs/data/blob/3016d6d0e8bdff85995933a69f7162c1ff081734/src/utils/removeInternalProperties.ts#L10-L23

Do you think they'd cover your intentions here? If not, I'd like to learn more why (based on the function names, their intentions are the same).

Reply:
Unfortunately they do different things. Well, maybe similar but in different steps of the process.
Ofc I'm totally open to provide a different name for `isRelationalProperty` or implement it in a different way. But I wasn't able to make the suggested change work.

For example, in the `defineRelationalProperties` function with the following `entity` input (
```
    {
      id: '3ff81810-2161-457a-9b1a-2219a73db387',
      name: 'user',
      info: { firstName: 'Ryuichi', lastName: 'Sakamoto' },
      __type: 'user',
      __primaryKey: 'id'
    }
```
the `properties` object is constructed this way
```
properties: { country: { enumerable: true, get: [Function: get] } },
```
and, given that `Object.defineProperties` won't work for nested properties, I had to setup a `defineNestedProperties` function.